### ISSUE TITLE
Feature Request: Scheduled ADMIN_DOWN for Servers

### ISSUE THREAD

**Issue by MattMills** [apache/incubator-trafficcontrol#1303]
If it was possible for an Admin user to add a "maintenance" event to a server (Start time, stop time , required reason, username), that data could then be used so that when the server runs ORT within that time period the actual server status (if status == reported)  is overridden with ADMIN_DOWN. 

This would enable ops to pre-program the network maintenance and other stuff we need to work around, as well as give us a fantastic way (much more visible) to keep track of it for troubleshooting purposes.

**Comment by mitchell852** [comment]
@dneuman64 @jhg03a - is this still required with the implementation of https://github.com/apache/trafficcontrol/blob/master/blueprints/multi-interface-servers.md

**Comment by jhg03a** [comment]
Server interfaces doesn't address this request

**Comment by MattMills** [comment]
I agree with Jonathan, the intention is to provide a mechanism by which TC itself can be "aware" of expected periods of downtime and route around them with no other operator intervention.

**Comment by mitchell852** [comment]
> Server interfaces doesn't address this request

agreed. just wondered if a server having multiple interfaces reduced the need to even admin down servers for maintenance purposes. i guess not.

**Comment by jhg03a** [comment]
This ticket is predicated on queue and snap being safe for automation to perform.
